internal/driver/controllerserver_helper.go

```diff
@@ -719,13 +665,17 @@ func (cs *ControllerServer) attachVolume(ctx context.Context, volumeID, linodeID
 		PersistAcrossBoots: &persist,
 	})
 	if err != nil {
-		code := codes.Internal // Default error code is Internal.
-		// Check if the error indicates that the volume is already attached.
+		// https://github.com/container-storage-interface/spec/blob/master/spec.md#controllerpublishvolume-errors
 		var apiErr *linodego.Error
-		if errors.As(err, &apiErr) && strings.Contains(apiErr.Message, "is already attached") {
-			code = codes.Unavailable // Allow a retry if the volume is already attached: race condition can occur here
+		if errors.As(err, &apiErr) {
+			switch {
+			case strings.Contains(apiErr.Message, "is already attached"):
+				return errAlreadyAttached
+			case strings.Contains(apiErr.Message, "Maximum number of block storage volumes are attached to this Linode"):
+				return errMaxAttachments
```

Comment:
Similar to my suggestion from other comment. We should try to pass along the err message by the api here too